devtron-labs/dashboard
# feat: Show/Hide Managed Fields UI from manifest

# Description

The managedFields config under the manifest takes up a larger view section and should be toggled by help of a checkbox.

Fixes [AB 1972](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=1972)

## Type of change


# Reference
reference of managedFields section of a manifest :

![image](https://user-images.githubusercontent.com/68757539/218189844-46405a0a-6fa7-458c-bf58-2de6e4bea2f0.png)

# Expected UX:
- Hide Managed Fields checkbox to toggle, i.e. to show/hide the managedFields section.
- It should be relevant and only be shown if :
  - Manifest tab is active.
  - ManagedFields exists under the selected manifest.
- On selected checkbox, the managedFields should be removed (not collapsed) from the manifest. 

![image](https://user-images.githubusercontent.com/68757539/218189926-38cb591a-0d5a-496c-860f-ce95d1021146.png)


# How Has This Been Tested?

- Go to any app details (must be deployed)
- Select any resource
- Open the manifest for that resource
- If managedFields exist, there should be a checkbox [Hide Managed Fields] to show/hide the managed fields

- [x] Tested for manifests having managedFields
- [x] Tested for manifests not having managedFields


# Checklist:
